packages/tuono/src/build/config/create-json-config.spec.ts

```diff
@@ -0,0 +1,32 @@
+import fs from 'fs/promises'
+
+import { describe, expect, it, vitest } from 'vitest'
+import react from '@vitejs/plugin-react-swc'
+
+import { createJsonConfig } from './create-json-config'
+
+const writeFileSpy = vitest.spyOn(fs, 'writeFile').mockResolvedValue(void 0)
+
+describe('createJsonConfig', () => {
+  const sampleConfig = { server: { host: 'h', port: 1 } }
+
+  it('should process config with only server property', async () => {
+    await createJsonConfig(sampleConfig)
+
+    expect(writeFileSpy).toHaveBeenCalledWith(
+      expect.any(String),
+      expect.stringContaining(JSON.stringify(sampleConfig)),
+      expect.any(String),
+    )
+  })
+
+  it('should process config with plugins', async () => {
+    await createJsonConfig({ ...sampleConfig, vite: { plugins: [react()] } })
```

Comment:
WDYT about removing the `react` plugin so that we are keeping it more framework-agnostic?

```suggestion
    await createJsonConfig({ ...sampleConfig, vite: { plugins: [{id: 'plugin'} as PluginOption] } })
```

Reply:
Using a simple JS object, like you proposed, won't cause any serialisation error.
Right now `react` / `tailwind` plugins return something that cannot be serialised,
which was the cause of the error which this test is aimed to prevent in the future.

Since we are not removing `react` soon I would consider investigating an alternative later.

Reply:
As far as I understood in the test we want to check just that vite is stripped out. There is no need of keeping the correct values

Reply:
No, this test is aimed to prevent regression of #414.
The issue was caused by non serialisable data inside the `vite.plugins`, 
so to avoid regression we must create the same scenario where the `vite.plugins` contains not serialisable data.

Reply:
What I mean is that you are not checking that the data is not serializable (and then you strip it out). You are just removing `vite` from the cloned value.
For this reason in the test I think you don't need to put non-serializable data within the `vite` object since it will always be stripped

Reply:
Right now there are no known data that be set to not serialisable values except `vite.plugins`.

Again, the point here is to prevent regression of #414...
The test recreates the exact scenario that was causing that issue.

Do the following steps:
1. Restore the previous code inside `removeViteProperties`
    ```ts
     const newConfig = structuredClone(config)
     delete newConfig['vite']
     return newConfig
     ```
3. Run`pnpm --filter=tuono run test` the test with `react` plugin. 
    You will experience a `DataCloneError` error (#414)

4. Replace the `react`  plugin with a plain object.
    No error is present. 

So, with your test code there is no error with the old implementation...
so I don't see any valid reason to write that code since it was working with the old implementation which was causing  #414 error

Reply:
I got it! Thanks